test/jdk/sun/security/provider/acvp/Launcher.java

```diff
@@ -0,0 +1,128 @@
+        //
+        // The files are either put into the "data" directory or another
+        // directory specified by the "test.acvp.data" test property.
+        // The test walks through the directory recursively and looks for
+        // file names equals to or ending with "internalProjection.json" and
+        // runs test on them.
+        //
+        // Set the "test.acvp.alg" test property to only test this algorithm.
+        //
+        // Sample files can be downloaded from
+        // https://github.com/usnistgov/ACVP-Server/tree/master/gen-val/json-files.
+        //
+        // By default, the test uses system-preferred implementations.
+        // If you want to test on a specific provider, set the
+        // "test.acvp.provider" test property. The provider must be
+        // registered.
+        //
+        // Tests for each algorithm must be compliant to its specification linked from
+        // https://github.com/usnistgov/ACVP?tab=readme-ov-file#supported-algorithms.
+        //
+        // Example:
+        //
+        // jtreg -Dtest.acvp.provider=SunJCE \
+        //       -Dtest.acvp.alg=ML-KEM \
+        //       -Dtest.acvp.data=/path/to/json-files/ \
+        //       -jdk:/path/to/jdk Launcher.java
+
+        var testDataProp = System.getProperty("test.acvp.data");
+        Path dataPath = testDataProp != null
+                ? Path.of(testDataProp)
+                : Path.of(System.getProperty("test.src"), "data");
+        System.out.println("Data path: " + dataPath);
+
+        if (PROVIDER != null) {
+            System.out.println("Provider: " + PROVIDER.getName());
+        }
+        if (ONLY_ALG != null) {
+            System.out.println("Algorithm: " + ONLY_ALG);
+        }
+
+        try (var stream = Files.walk(dataPath)) {
+            stream.filter(Files::isRegularFile)
+                    .filter(p -> p.getFileName().toString()
+                            .endsWith("internalProjection.json"))
+                    .forEach(Launcher::run);
+        }
+    }
+
+    static void run(Path test) {
+        try {
+            JSONValue kat;
+            try {
+                kat = JSONValue.parse(Files.readString(test));
+            } catch (Exception e) {
+                System.out.println("Warning: cannot parse " + test + ". Skipped");
+                return;
+            }
+            var alg = kat.get("algorithm").asString();
+            if (ONLY_ALG != null && !alg.equals(ONLY_ALG)) {
+                return;
+            }
+            System.out.println(">>> Testing " + test + "...");
+            switch (alg) {
+                case "ML-DSA" -> ML_DSA_Test.run(kat, PROVIDER);
+                case "ML-KEM" -> ML_KEM_Test.run(kat, PROVIDER);
+                case "SHA2-256", "SHA2-224", "SHA3-256", "SHA3-224"
+                    -> SHA_Test.run(kat, PROVIDER);
+                default -> System.out.println("Skipped unsupported algorithm: " + alg);
```

Comment:
This seems more like a test configuration error and should not silently be ignored (you have to read the output to see the error/warning).
Perhaps it could be a "Skipped" kind of test, so it gets reported.

Reply:
Since there is no test data now, the only way to run this test is to point `test.acvp.data` to an existing ACVP testsuite data directory and it might contain tests on other algorithms we don't support yet.